### PR TITLE
Sortings (c) Alexey

### PR DESCRIPTION
**Pull request by tarasevich**
@den-sergeyev , @egorhm , @sergeyenin 

Could you please take a look at these? Any observations or improvements?

Haven't actually read any theory.

### CODE REVIEW

File: Sorting/src/BubbleSort.java [hunk 1]
```diff
@@ -1,8 +1,18 @@
-/**
- * Created by Egor Hamaliy on 4/21/14.
- */
+import java.util.Arrays;
+
 public class BubbleSort implements Sortable {
     public int[] sort(int[] numbers) {
-        return null;
+        int[] a = Arrays.copyOf(numbers, numbers.length);
```
Comment (sergeyenin) [inline Sorting/src/BubbleSort.java:5]: what is the reason to copy input array instead of just modifying it?

File: Sorting/src/InsertionSort.java [hunk 2]
```diff
@@ -1,8 +1,25 @@
...
+
+        for (int i = 1; i <= l-1; ++i) { // index of x
+            int x = a[i];
+
+
+            int j = 0; // insertion position, stable
+            for (; a[j] <= x && j < i; ++j);
+
+            // rotate the range right
+            for (int k = i; k-1 >= j; --k)
```
Comment (sergeyenin) [inline Sorting/src/InsertionSort.java:18]: what is the reason of double passing through array(line 15 and 18)?
is there possibility to merge both cycles into single one?

File: Sorting/src/BubbleSort.java (hunk 1 above)
Comment (tarasevich) [inline Sorting/src/BubbleSort.java:5]: if you return a result, you should not modify the arguments.

File: Sorting/src/InsertionSort.java (hunk 2 above)
Comment (tarasevich) [inline Sorting/src/InsertionSort.java:18]: it's single passing. it's just two arrays.

generally, you are right. it's possible to eliminate the first loop.

File: Sorting/src/BubbleSort.java (hunk 1 above)
Comment (sergeyenin) [inline Sorting/src/BubbleSort.java:5]: why?
specifically, why you not should return sorted array from function which is designed to sort the input array?

**Pull request merged**